prog/postgres/postgres_server_nexus.rb

```diff
@@ -179,8 +181,7 @@ def before_run
     vm.sshable.cmd("sudo -u postgres psql", stdin: commands)
 
     when_initial_provisioning_set? do
-      decr_initial_provisioning
-      push self.class, frame, "restart"
+      bud self.class, frame, "restart"
```

Comment:
I initially used push, however when it pops, it re-run this state and re-push restart, because we don't decrement the initial_provisioning semaphore anymore. I multiple decr_initial_provisioning to wait for centralizing it.

Restart in this case is quite fast because the database is just created and there is no WAL file to be replayed (which is the usual cause of long running restarts), so I thought it is OK to be wrong for few seconds.

Alternative would be using push and calling decr_initial_provisioning as well, but then we unnecessarily update the superuser_password again (or we would add an extra check there), which is not horrible but annoying.

Reply:
If it is quick I'm okay with `bud`. 

We don't have many examples of `push`, but I believe the method to prevent the label rerun involves checking `retval` first in the label `hop_wait if retval&.dig("msg") == "postgres server is restarted"`. `Prog::BootstrapRhizome` provides this example.

Reply:
yeah, push was basically something of an experiment when it was last thought over. It just seemed like if there's a stack there should be a push. But the way returns work is really weird and error prone. Jury's out on whether push is even a good idea or if maybe stacks were bad but supervision trees (bud) were good.

Reply:
I think we can give more chance to push for a while. It has interesting use cases. Maybe we can iron out the weird parts.
Same for stacks, we rarely use them, most of the time our stacks are of depth 1, but are we sure we wouldn't need them at all.

Reply:
it did occur to me that perhaps push should be more like "call with continuation" rather than picking up the retval and branching on that in the state, e.g. it would know where to go next upon "pop". That may be one way to fix it.

Reply:
also the "bud" interface is pretty brutal, it'd be nice to have `self.class.bud` so it can fill in at least the class field, or even `self.class.bud_labelname.` To avoid freehanding so many identifiers (much as semaphores are less free-hand than the last several systems via a bit of code generation)

Reply:
Call with continuation would be quite nice actually.

Reply:
I opened an issue for it; https://github.com/ubicloud/ubicloud/issues/1187